anabot/runtime/installation/hub/partitioning/specialized_disks/__init__.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+import logging
+logger = logging.getLogger('anabot')
+
+from fnmatch import fnmatchcase
+
+from anabot.runtime.decorators import handle_action, handle_check, check_action_result
+from anabot.runtime.default import default_handler
+from anabot.runtime.functions import get_attr, getnode, getnodes, getsibling
+from anabot.runtime.errors import TimeoutError
+from anabot.runtime.translate import tr
+from anabot.runtime.installation.common import done_handler
+from anabot.runtime.actionresult import NotFoundResult as NotFound, ActionResultPass as Pass, ActionResultFail as Fail
+
+_local_path = '/installation/hub/partitioning/add_spec_disk'
+def handle_act(path, *args, **kwargs):
+    return handle_action(_local_path + path, *args, **kwargs)
+
+def handle_chck(path, *args, **kwargs):
+    return handle_check(_local_path + path, *args, **kwargs)
+
+def select_manipulate(element, app_node, local_node, dryrun):
+    name = get_attr(element, "names", "*")
+    action = get_attr(element, "action", "select")
+
+    try:
+        results_label = getnode(app_node, "label", tr("Search Res_ults:", context="GUI|Installation Destination|Filter|Search"))
+        table_pane = getsibling(results_label, 1, "scroll pane")
+        disks_table = getnode(table_pane, "table")
+    except TimeoutError as e:
+        return NotFound("Results found label, table pane or disks table: %s" % e)
+    cells = getnodes(disks_table, "table cell")
+    # every table row has 12 cells, with checkbox on position 0 and device name on 1
```

Comment:
Is it possible to avoid hardcoding the 12? Like detecting number of entries in the table and dividing number of cells by the number of entries?

Reply:
At first I wasn't really sure how to achieve this (as there is just a bunch of table cells, without sections of rows. But now after your suggestion I realized there are the table header cells which can be used to count the number of cells in a row, therefore making possible to get rid of the magic '12' constant.